src/App.tsx

```diff
@@ -15,48 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface Good {
+  name: string;
+  id: number;
+}
+
+enum SortType {
+  Name = 'name',
+  Length = 'length',
+  None = 'no-sorting',
+}
+
+export const goodsObject: Good[] = goodsFromServer
+  .map((good, index) => ({ name: good, id: index }));
+
+function getSortedGoods(
+  goods: Good[],
+  sortType: SortType,
+  toReverse: boolean,
+): Good[] {
+  const sortedGoods = [...goods];
+
+  if (sortType === SortType.Name) {
+    sortedGoods.sort((good1, good2) => good1.name.localeCompare(good2.name));
+  }
+
+  if (sortType === SortType.Length) {
+    sortedGoods.sort((good1, good2) => good1.name.length - good2.name.length);
+  }
+
+  return toReverse ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getSortedGoods(goodsObject, sortType, isReversed);
+  const isResetSeen = sortType !== SortType.None || isReversed;
+
+  const handleReverseButton = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const handleReset = () => {
+    setSortType(SortType.None);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info',
+            { 'is-light': sortType !== SortType.Name })}
+          onClick={() => setSortType(SortType.Name)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success',
+            { 'is-light': sortType !== SortType.Length })}
+          onClick={() => setSortType(SortType.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning',
+            { 'is-light': !isReversed })}
+          onClick={handleReverseButton}
         >
           Reverse
         </button>
```

Comment:
it is a perfect case when you create array with objects which represents every button and just map them.
Including onClick functions, classes, text, etc.